test/metabase/pivot/core_test.cljc

```diff
@@ -0,0 +1,273 @@
+           :field_ref ["aggregation" 0],
+           :effective_type "type/BigInteger",
+           :aggregation_index 0,
+           :ident "S75TeXxnjAh0tJJDbzKg2",
+           :display_name "Count",
+           :remapping nil,
+           :remapped_from_index nil,
+           :base_type "type/BigInteger"}]})
+
+(deftest columns-without-pivot-group-test
+  (testing "Correctly filters out the pivot grouping column based on name"
+    (is (= ["col0" "col1" "col2" "count"]
+           (->> (pivot/columns-without-pivot-group (:cols pivot-test-data))
+                (map :name))))))
+
+(deftest split-pivot-data
+  (testing "split-pivot-table pulls apart the aggregations packed into a single
+    result set, keyed by the columns indexes that are aggregated"
+    (is (= {:pivot-data {[0 1 2] [[1 "A" "Y" 1]
+                                  [1 "A" "Z" 1]
+                                  [1 "B" "Y" 1]
+                                  [1 "B" "Z" 1]
+                                  [1 "C" "Y" 1]
+                                  [1 "C" "Z" 1]
+                                  [2 "A" "Y" 1]
+                                  [2 "A" "Z" 1]
+                                  [2 "B" "Y" 1]
+                                  [2 "B" "Z" 1]
+                                  [2 "C" "Y" 1]
+                                  [2 "C" "Z" 1]]
+                         [1 2]   [[nil "A" "Y" 2]
+                                  [nil "A" "Z" 2]
+                                  [nil "B" "Y" 2]
+                                  [nil "B" "Z" 2]
+                                  [nil "C" "Y" 2]
+                                  [nil "C" "Z" 2]]
+                         [2]     [[nil nil "Y" 6]
+                                  [nil nil "Z" 6]]
+                         [0 1]   [[1 "A" nil 2]
+                                  [1 "B" nil 2]
+                                  [1 "C" nil 2]
+                                  [2 "A" nil 2]
+                                  [2 "B" nil 2]
+                                  [2 "C" nil 2]]
+                         [1]     [[nil "A" nil 4]
+                                  [nil "B" nil 4]
+                                  [nil "C" nil 4]]
+                         []      [[nil nil nil 12]]}
+            :primary-rows-key [0 1 2]}
+           ;; Dissoc :columns beacuse this is the same as the result of `columns-without-pivot-group-test`, tested above
+           (dissoc (pivot/split-pivot-data pivot-test-data) :columns)))))
+
+(deftest get-subtotal-values-test
+  (testing "Extracts subtotal values from pivot data"
+    (let [pivot-data {[0 1 2] [[1 "A" "Y" 10]
+                               [1 "B" "Z" 20]]}
+          val-indexes [3]
+          result (#'pivot/get-subtotal-values pivot-data val-indexes nil)]
+      (is (= {[0 1 2] {[1 "A" "Y"] [10]
+                       [1 "B" "Z"] [20]}}
+             result))))
+
+  (testing "Excludes the primary rows if passed a primary rows key"
```

Comment:
could you bust up the inner testing forms? it makes it easier to run one at a time and track down test failures